spec/requests/private_controller_spec.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe PrivateController, type: :controller do
+  describe 'GET#index' do
+    context 'when user is not sign in' do
+      it 'returns redirect status' do
+        get :index
+
+        expect(response).to have_http_status(:found)
+      end
+    end
+
+    context 'when user is sign in' do
+      before(:each) do
+        sign_in Fabricate(:user)
+      end
```

Comment:
Please check this convention: https://nimblehq.co/compass/development/code-conventions/ruby/rspec/#fixtures-isolation

We don't use the `before` and `after` when writing a spec.

Reply:
Thank you for pointing this out. Could I ask for more clarification on the why of this conventions, is it because most test files are big and having the `before` `after` make the tests more unclear?

Reply:
yeah, tbh i am not propose this convention, and not sure the correct main idea behind it. 😅  But i think to avoid the global definition in when writing a spec. (as the heading is `Fixtures isolation`)

Every `single context` should define the related fixtures (test data) within itself. We don't have to look up up and down when debugging a specific test, no surprises 😮 It's not actually DRY however i think this is a good trade-off.

cc @malparty what is you idea on this?